### ISSUE TITLE
Correct explanation of why `readonly` only affects text controls

### ISSUE THREAD

**Issue by aaronadamsCA** [mdn/content#27408]
### MDN URL

https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/readonly

### What specific section or headline is this issue about?

Overview

### What information was incorrect, unhelpful, or incomplete?

> Note: Only text controls can be made read-only, since for other controls (such as checkboxes and buttons) there is no useful distinction between being read-only and being disabled, so the readonly attribute does not apply.

### What did you expect to see?

Something more like this:

> It's important to understand that READONLY merely prevents the user from changing the value of the field, not from interacting with the field. For many types of fields, READONLY is irrelevent because you don't normally change the value. In checkboxes, for example, you can check them on or off (thus setting the CHECKED state) but you don't change the value of the field. DISABLED, however, actually prevents you from using the field.
>
> _Source: http://www.faqs.org/docs/htmltut/forms/_INPUT_DISABLED.html_

### MDN metadata

<details>
<summary>Page report details</summary>

* Folder: `en-us/web/html/attributes/readonly`
* MDN URL: https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/readonly
* GitHub URL: https://github.com/mdn/content/blob/main/files/en-us/web/html/attributes/readonly/index.md
* Last commit: https://github.com/mdn/content/commit/942a529383ee7ee3996fb234187641c08935f3ff
* Document last modified: 2023-04-13T06:13:15.000Z

</details>

**Comment by Josh-Cena** [comment]
This note is directly copied from the [HTML standard](https://html.spec.whatwg.org/#the-readonly-attribute), so I'll take it as more technically accurate than a website from the 2000s. Is there anything fundamentally incompatible between their content?

**Comment by aaronadamsCA** [comment]
I think the `readonly` spec might be wrong, actually? It says "when specified, the element is not [mutable](https://html.spec.whatwg.org/#concept-fe-mutable)." It defines mutability as "whether or not the user can modify the value or checkedness of a form control." But `readonly` does not affect the user's ability to modify checkedness.

I think it would be more correct if both the HTML spec and the MDN doc clearly stated that `readonly` controls whether or not the user can change the `value` of a text control.

And then I think a more helpful clarification in the MDN doc would be that the `readonly` attribute is only useful on text controls because it only prevents the user from modifying `value`, not checkedness; and that as a result, a user can still effectively change the submitted value of a readonly radio button or checkbox control by changing its checkedness.

I hope that makes sense.

**Comment by Josh-Cena** [comment]
Could you also send an issue to the HTML spec and see what they think? I'm okay to include more information but I don't want MDN and HTML to be _different_.

**Comment by aaronadamsCA** [comment]
Good idea, I just filed https://github.com/whatwg/html/issues/9496 to see what they say. Thanks!
